scripts/zkai_sota_artifact_watchlist_gate.py

```
@@ -0,0 +1,938 @@
+                    "public_verifier_input_available",
+                    "baseline_verification_reproducible",
+                    "metadata_mutation_reproducible",
+                )
+            ):
+                raise SotaWatchlistError(f"empirical adapter row lacks reproducible artifacts for {system}")
+            if not row["local_evidence"]:
+                raise SotaWatchlistError(f"empirical adapter row missing local evidence for {system}")
+            _resolve_repo_relative_existing_file(row["local_evidence"], f"local evidence for {system}")
+        if system in SOURCE_CONTEXT_ONLY_SYSTEMS:
+            if status == STATUS_EMPIRICAL:
+                raise SotaWatchlistError(f"source-backed system promoted to empirical adapter: {system}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)